tracks/python/exercises/kindergarten-garden/mentoring.md

```diff
@@ -16,21 +16,25 @@ class Garden(object):
         'V' : 'Violets'
     }
 
-    def __init__(self, diagram, students=DEFAULT_STUDENTS):
-        self.rows = diagram.split('\n')
-        self.students = sorted(students)
+    def __init__(self, diagram, students=None):
+        self.rows = diagram.splitlines()
+        self.students = sorted(students if students is not None else self.DEFAULT_STUDENTS)
+        self.student_index = {name:index for (index, name) in enumerate(self.students)}
```

Comment:
An index lookup is an interesting approach, but `Garden.plants` will die with a somewhat confusing IndexError if a student who _is_ a member of the class but wasn't assigned a garden is requested (ie **diagram** shorter than **students**). In my mind requesting a valid student who didn't get a garden assigned should return an empty list, while a student who isn't valid should either do the same or potentially throw a KeyError. I'll admit though that what the class is actually _supposed_ to do in this case isn't well defined in the exercise. As is what should be done if the diagram exceeds the number of students.